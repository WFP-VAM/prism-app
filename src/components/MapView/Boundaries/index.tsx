import React from 'react';
import { get } from 'lodash';
import { GeoJSONLayer } from 'react-mapbox-gl';
import * as MapboxGL from 'mapbox-gl';
import adminBoundaries from '../../../config/admin_boundaries.json';

const linePaint: MapboxGL.LinePaint = {
  'line-color': 'grey',
  'line-width': 1,
  'line-opacity': 0.3,
};

/**
 * To activate fillOnClick option, we "fill in"
 * polygons with opacity 0.
 */
const fillPaint: MapboxGL.FillPaint = { 'fill-opacity': 0 };

// Get admin data to process.
function getAdminData(evt: any) {
  // eslint-disable-next-line
  console.log(get(evt.features[0], 'properties.ADM2_PCODE'));
}

function Boundaries() {
  return (
    <GeoJSONLayer
      data={adminBoundaries}
      linePaint={linePaint}
      fillPaint={fillPaint}
      fillOnClick={(evt: any) => {
        getAdminData(evt);
      }}
    />
  );
}

export default Boundaries;
