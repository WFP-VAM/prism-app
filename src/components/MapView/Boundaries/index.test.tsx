import React from 'react';
import ReactMapboxGl from 'react-mapbox-gl';
import { render } from '@testing-library/react';

import Boundaries from '.';

const Map = ReactMapboxGl({
  accessToken: 'TOKEN',
});

test('renders as expected', () => {
  const { container } = render(
    <Map
      // eslint-disable-next-line react/style-prop-object
      style="mapbox://styles/mapbox/light-v10"
      center={[90, 90]}
      zoom={[5]}
      containerStyle={{
        height: '100vh',
        width: '100vw',
      }}
    >
      <Boundaries />
    </Map>,
  );
  expect(container).toMatchSnapshot();
});
