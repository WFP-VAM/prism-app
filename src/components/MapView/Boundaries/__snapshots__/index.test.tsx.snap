// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div>
  <div
    style="height: 100vh; width: 100vw;"
  />
</div>
`;
